### ISSUE TITLE
Coverted Tensorflow2 saved model to ONNX model returns Unknown C++ exception from OpenCV code

### ISSUE THREAD

**Issue by SEHAIRIKamal** [onnx/tensorflow-onnx#1906]
Hi all, 
I am trying to convert my exported TensorFlow 2 model trained on my own dataset to onnx format using tf2onnx . the aim is to use it with OpenCV dnn.readNetFromONNX function. For Conversion, I used the following code since I am using TensorFlow 2  
`python -m tf2onnx.convert --saved-model tensorflow-model-path --output model.onnx --opset 15`
The conversion process generates successfully the model.onnx and returns the following:
`n -m tf2onnx.convert --saved-model C:/Tensorflow/workspace/training_demo/exported-models/my_model/saved_model --output model.onnx --opset 15
C:\Tensorflow\venv\lib\site-packages\numpy\_distributor_init.py:30: UserWarning: loaded more than 1 DLL from .libs:
C:\Tensorflow\venv\lib\site-packages\numpy\.libs\libopenblas.WCDJNK7YVMPZQ2ME2ZZHJJRJ3JIKNDB7.gfortran-win_amd64.dll
C:\Tensorflow\venv\lib\site-packages\numpy\.libs\libopenblas.XWYDX2IKJW2NMTWSFYNGFUWKQU3LYTCZ.gfortran-win_amd64.dll
  warnings.warn("loaded more than 1 DLL from .libs:"
C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python39_64\lib\runpy.py:127: RuntimeWarning: 'tf2onnx.convert' found in sys.modules after import of package 'tf2onnx', but prior to execution of 'tf2onnx.convert'; this may result in unpredictable behaviour
  warn(RuntimeWarning(msg))
2022-04-07 16:58:02,939 - WARNING - '--tag' not specified for saved_model. Using --tag serve
2022-04-07 16:58:13,132 - INFO - Signatures found in model: [serving_default].
2022-04-07 16:58:13,133 - WARNING - '--signature_def' not specified, using first signature: serving_default
2022-04-07 16:58:13,137 - INFO - Output names: ['detection_anchor_indices', 'detection_boxes', 'detection_classes', 'detection_multiclass_scores', 'detection_scores', 'num_detections', 'raw_detection_boxes', 'raw_detection_scores']
WARNING:tensorflow:From C:\Tensorflow\venv\lib\site-packages\tf2onnx\tf_loader.py:711: extract_sub_graph (from tensorflow.python.framework.graph_util_impl) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.compat.v1.graph_util.extract_sub_graph`
2022-04-07 16:58:18,989 - WARNING - From C:\Tensorflow\venv\lib\site-packages\tf2onnx\tf_loader.py:711: extract_sub_graph (from tensorflow.python.framework.graph_util_impl) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.compat.v1.graph_util.extract_sub_graph`
2022-04-07 16:58:22,197 - INFO - Using tensorflow=2.8.0, onnx=1.11.0, tf2onnx=1.10.0/07e9e0
2022-04-07 16:58:22,198 - INFO - Using opset <onnx, 15>
2022-04-07 16:58:29,140 - INFO - Computed 0 values for constant folding
2022-04-07 16:58:29,144 - INFO - Computed 0 values for constant folding
2022-04-07 16:58:33,926 - INFO - Computed 4 values for constant folding
2022-04-07 16:58:40,244 - INFO - folding node using tf type=ExpandDims, name=StatefulPartitionedCall/Postprocessor/ExpandDims_2
2022-04-07 16:58:40,245 - INFO - folding node using tf type=Identity, name=StatefulPartitionedCall/MultiscaleGridAnchorGenerator/GridAnchorGenerator/Identity
2022-04-07 16:58:40,248 - INFO - folding node using tf type=Select, name=StatefulPartitionedCall/Postprocessor/BatchMultiClassNonMaxSuppression/PadOrClipBoxList/Select_1
2022-04-07 16:58:40,252 - INFO - folding node using tf type=Select, name=StatefulPartitionedCall/Postprocessor/BatchMultiClassNonMaxSuppression/PadOrClipBoxList/Select_8
2022-04-07 16:58:40,291 - INFO - folding node type=Range, name=StatefulPartitionedCall/Postprocessor/range
2022-04-07 16:58:43,261 - INFO - Optimizing ONNX model
2022-04-07 16:58:57,710 - INFO - After optimization: BatchNormalization -94 (97->3), Cast -60 (125->65), Const -913 (1196->283), Identity -76 (76->0), Less -1 (5->4), Mul -2 (48->46), Placeholder -2 (6->4), ReduceSum -1 (2->1), Reshape -22 (46->24), Shape -2 (17->15), Slice -9 (39->30), Split -2 (15->13), Squeeze -11 (53->42), Sub -1 (26->25), Transpose -401 (427->26), Unsqueeze -16 (59->43)
2022-04-07 16:58:58,654 - INFO -
2022-04-07 16:58:58,654 - INFO - Successfully converted TensorFlow model C:/Tensorflow/workspace/training_demo/exported-models/my_model/saved_model to ONNX
2022-04-07 16:58:58,656 - INFO - Model inputs: ['input_tensor']
2022-04-07 16:58:58,658 - INFO - Model outputs: ['detection_anchor_indices', 'detection_boxes', 'detection_classes', 'detection_multiclass_scores', 'detection_scores', 'num_detections', 'raw_detection_boxes', 'raw_detection_scores']
2022-04-07 16:58:58,661 - INFO - ONNX model is saved at model.onnx`

However, when I try to read the converted model I get the following error:
`File "C:\Tensorflow\testcovertedTF2ToONNX.py", line 10, in <module>
    net = cv2.dnn.readNetFromONNX('C:/Tensorflow/model.onnx')
cv2.error: Unknown C++ exception from OpenCV code`

Please, can someone help resolve this error. 

**System information**
- OS Platform: Windows 10 64 bits
- Tensorflow Version: 2.8
- Python version: 3.9.7
- OpenCV version: 4.5.5

**To Reproduce**
testcovertedTF2ToONNX.py
```
import cv2
import numpy as np
 

image = cv2.imread("img002500.jpg")
if image is None:
	print("image emplty")
image_height, image_width, _ = image.shape
net = cv2.dnn.readNetFromONNX('model.onnx')
image = image.astype(np.float32)
# create blob from image (opencv dnn way of pre-processing)
input_blob = cv2.dnn.blobFromImage(image, 1, (640,640), 0, swapRB=False, crop=False)
net.setInput(input_blob)
output = net.forward()
# loop over each of the detection
for detection in output[0, 0, :, :]:
   # extract the confidence of the detection
   confidence = detection[2]
   # draw bounding boxes only if the detection confidence is above...
   # ... a certain threshold, else skip
   if confidence > .4:
       # get the class id
       class_id = detection[1]
       # map the class id to the class
       class_name = 'person'
       # get the bounding box coordinates
       box_x = detection[3] * image_width
       box_y = detection[4] * image_height
       # get the bounding box width and height
       box_width = detection[5] * image_width
       box_height = detection[6] * image_height
       # draw a rectangle around each detected object
       cv2.rectangle(image, (int(box_x), int(box_y)), (int(box_width), int(box_height)), (255,0,0), thickness=2)
       # put the FPS text on top of the frame
       cv2.putText(image, class_name, (int(box_x), int(box_y - 5)), cv2.FONT_HERSHEY_SIMPLEX, 1, (255,0,0), 2)
 
cv2.imshow('image', image)
cv2.imwrite('image_result.jpg', image)
cv2.waitKey(0)
cv2.destroyAllWindows()
```

**Screenshots**
![0](https://user-images.githubusercontent.com/23197960/162252422-710e4e5d-4c0a-4a1c-acb7-179093768efd.jpg)

**saved model**
You can find the saved model by downloading the zip file below
https://www.dropbox.com/s/9exu9c0dqgahenl/saved_model.zip?dl=0

**Comment by fatcat-z** [comment]
I tried to export the onnx file and check them by onnx checker, it succeeded:

import onnx
onnx_model = onnx.load("model.onnx")
onnx.checker.check_model(onnx_model)

And if I run "net = cv2.dnn.readNetFromONNX('model.onnx')" code with cv2 4.5.5, it will break without any errors. But if we run the code with cv2 4.5.1 version, some details will be thrown:

cv2.error: OpenCV(4.5.1) /tmp/pip-req-build-ms668fyv/opencv/modules/dnn/src/onnx/onnx_importer.cpp:1887: error: (-2:Unspecified error) in function 'handleNode'
> Node [Less]:(__inference_map_while_cond_8475_6769_map/while/Less:0) parse error: OpenCV(4.5.1) /tmp/pip-req-build-ms668fyv/opencv/modules/dnn/src/dnn.cpp:614: error: (-2:Unspecified error) Can't create layer "__inference_map_while_cond_8475_6769_map/while/Less:0" of type "Less" in function 'getLayerInstance'

Could you please prepare the inputs and then pass it to ONNXRuntime to see if it can inference successfully? Also, you can try to download OpenCV code and do a local debug.

**Comment by fatcat-z** [comment]
Closing this. Feel free to open a new one if the issue still exists.

**Comment by SEHAIRIKamal** [comment]
> I tried to export the onnx file and check them by onnx checker, it succeeded:
> 
> import onnx onnx_model = onnx.load("model.onnx") onnx.checker.check_model(onnx_model)
> 
> And if I run "net = cv2.dnn.readNetFromONNX('model.onnx')" code with cv2 4.5.5, it will break without any errors. But if we run the code with cv2 4.5.1 version, some details will be thrown:
> 
> cv2.error: OpenCV(4.5.1) /tmp/pip-req-build-ms668fyv/opencv/modules/dnn/src/onnx/onnx_importer.cpp:1887: error: (-2:Unspecified error) in function 'handleNode'
> 
> > Node [Less]:(__inference_map_while_cond_8475_6769_map/while/Less:0) parse error: OpenCV(4.5.1) /tmp/pip-req-build-ms668fyv/opencv/modules/dnn/src/dnn.cpp:614: error: (-2:Unspecified error) Can't create layer "__inference_map_while_cond_8475_6769_map/while/Less:0" of type "Less" in function 'getLayerInstance'
> 
> Could you please prepare the inputs and then pass it to ONNXRuntime to see if it can inference successfully? Also, you can try to download OpenCV code and do a local debug.

Hi, 
Thanks a lot for your reply.
I successfully ran the model using ONNXRuntime, but not with OpenCV.
The OpenCV DNN module (4.5 and <) is not compatible with TF2.
